replica/table.cc

```diff
@@ -119,7 +119,12 @@ table::make_sstable_reader(schema_ptr s,
     }
 }
 
-lw_shared_ptr<sstables::sstable_set> compaction_group::make_compound_sstable_set() const {
+lw_shared_ptr<sstables::sstable_set> compaction_group::make_sstable_set() const {
+    // Bypasses compound set if maintenance one is empty. If a SSTable is added later into maintenance,
+    // the sstable set is refreshed to reflect the current state.
```

Comment:
What do you mean by "refresh"? Further calls to `make_sstable_set()` will return a compound set? Or the already returned sstable sets are also upgraded?

Reply:
Any change to sstable set will cause this to be called again, as the set is immutable